From 95cdef9f5814480b783d7ad7e9197d36d07202c8 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 5 Dec 2014 16:28:47 +0800
Subject: [PATCH 267/269] ov2722: Remove sub device from platform data when
 unloading

This patch removes the v4l2 sub device of OV2722
from platform data when driver is unloading.

Change-Id: Ib1471b97ebbea1d3e10eb90dcee50dea6ca3de7c
Tracked-On:https://jira01.devtools.intel.com/browse/IMINAN-12256
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/31700
Integrator: Wei Tao <wei.tao@intel.com>
Reviewed-by: Jing Tao <jing.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov2722.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
index 8e5f208..41cd21c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
@@ -1412,6 +1412,9 @@ static int ov2722_remove(struct i2c_client *client)
 	dev->platform_data->csi_cfg(sd, 0);
 	v4l2_ctrl_handler_free(&dev->ctrl_handler);
 	v4l2_device_unregister_subdev(sd);
+#ifdef CONFIG_GMIN_INTEL_MID
+	atomisp_gmin_remove_subdev(sd);
+#endif
 	media_entity_cleanup(&dev->sd.entity);
 	kfree(dev);
 
-- 
1.9.1

